samanthas_telegram_bot/bot_data_loader/bot_data_loader.py

```diff
@@ -0,0 +1,236 @@
+                            AssessmentQuestionOption(**option) for option in question["options"]
+                        ),
+                    )
+                    for question in item["questions"]
+                ),
+            )
+            for item in data
+        )
+
+        # Each assessment has a sequence of age range IDs. We must match every single age range ID
+        # in this sequence to a respective assessment.
+        assessment_for_age_range_id = {
+            age_range_id: assessment
+            for assessment in assessments
+            for age_range_id in assessment.age_range_ids
+        }
+
+        return assessment_for_age_range_id
+
+    @classmethod
+    def _get_day_and_time_slots(cls) -> tuple[DayAndTimeSlot, ...]:
+        """Gets day and time slots from the backend."""
+
+        def get_hour(str_: str) -> int:
+            """Takes a string like 05:00:00 and returns hours (5 in this example)."""
+            return int(str_.split(":")[0])
+
+        data = cls._get_json(url_infix=API_URL_INFIX_DAY_AND_TIME_SLOTS)
+
+        return tuple(
+            DayAndTimeSlot(
+                id=item["id"],
+                day_of_week_index=item["day_of_week_index"],
+                from_utc_hour=get_hour(item["time_slot"]["from_utc_hour"]),
+                to_utc_hour=get_hour(item["time_slot"]["to_utc_hour"]),
+            )
+            for item in data
+        )
+
+    @classmethod
+    def _get_languages_and_levels(cls) -> tuple[LanguageAndLevel, ...]:
+        """Gets languages and levels from the backend."""
+
+        data = cls._get_json(
+            url_infix=API_URL_INFIX_LANGUAGES_AND_LEVELS,
+            name_for_logger="combinations of languages and levels",
+        )
+
+        return tuple(
+            LanguageAndLevel(
+                id=item["id"], language_id=item["language"]["id"], level=item["level"]
+            )
+            for item in data
+        )
+
+    @staticmethod
+    def _load_phrases() -> dict[str, MultilingualBotPhrase]:
+        """Reads bot phrases from CSV file, returns dictionary with internal IDs as key,
+        and a subclass of `TypedDict` as value, matching locales to actual phrases.
+        """
+
+        logger.info("Loading bot phrases")
+        with (Path(__file__).parent.resolve() / "phrases.csv").open(
+            encoding="utf-8", newline=""
+        ) as f:
+            reader = typing.cast(typing.Iterator[dict[str, str]], csv.DictReader(f))
+            return {
+                row["internal_id"]: MultilingualBotPhrase(
+                    **{
+                        locale: row[locale].replace("\\n", "\n")  # type: ignore[typeddict-item]
+                        for locale in LOCALES
+                    }
+                )
+                for row in reader
+            }
+
+    @staticmethod
+    def _get_json(
+        url_infix: str,
+        name_for_logger: str | None = None,
+        params: dict[str, str] | None = None,
+    ) -> Any:
```

Comment:
Is it really Any? Json is quite limited

Reply:
I know, but I get dozens of complaints from mypy otherwise :)